### ISSUE TITLE
Evolution chains are missing Alola form data

### ISSUE THREAD

**Issue by mousep** [PokeAPI/pokeapi#491]
The current database seems missing evolution chains data of Alola forms.

For example, query the evolution chain of Sandshrew:  
`https://pokeapi.co/api/v2/evolution-chain/11/`  
Only show that it will evolve to Sandslash on level 22, but missing data that the Alola form needs to use Ice Stone to evolve to Sandslash-Alola.

Maybe the current API interface is also difficult to store this information ?

**Comment by Kronopt** [comment]
Yeah, I don't think we have that data for evolution chains...
See if #455 is relevant to your question, though

**Comment by Naramsim** [comment]
@mousep, Can you please try here and tell us if those APIs provide what you are looking for? 

https://pokeapi-test-b6137.firebaseapp.com/api/v2

**Comment by Achtuur** [comment]
Bumping this quite a few months later, I hope that's not issue.

I tested out the pokeapi, the staging-pokeapi @Naramsim linked and Veekun and found that special evolution locations are missing for the Alola region (and also for the Hoenn region from ORAS).
The following locations are missing:
    - Alola's Lush Jungle, id 750 (contains a moss rock)
    - Alola's Mount Lanakila, id 775 (contains ice rock)
    - Alola's Blush Mountain (only USUM), id 771 (contains special magnetic zone)
    - Alola's Vast Poni Canyon, id 789 (contains special magnetic zone)
 (and also
    - Hoenn's (ORAS) Petalburg Woods (contains moss rock)
    - Hoenn's (ORAS) Shoal Cave (contains ice rock)
)
Leafeon and Glaceon do not have an entry in their evolution_chain.evolves_to.details for Lush Jungle and Mount Lanakila respectively. The pokemons that require a special magnetic zone, which are Magnezone, Vikavolt and Probopass do have an entry in their evolution_chain details but the location is null. They also require a new entry to account for eithr Blush Mountain or Vast Poni Canyon.

To summarise, Leafeon and Glaceon need a new entry in their species.evolution_chain.evolves_to.details which contains location id=750 and id=775 respectively, while Magnezone, Vikavolt and Probopass need their evolution details updated with locations id=771 and id=789
